File: examples/api/count_up/count_up.py
#! /usr/bin/python3
# -*- coding: utf-8 -*-
#
#	count_up.py
#
#						Jun/01/2015
#
#
# ---------------------------------------------------------------------
import	sys
import	json
import	cgi
#
sys.path.append ('/var/www/data_base/common/python_common')
from file_io import file_to_str_proc
from file_io import file_write_proc
# ---------------------------------------------------------------------
#
message = []
#
message.append ('*** count_up.py *** start ***')
#
form = cgi.FieldStorage()
message.append ('*** bbbb ***')
#
for key in form.keys():
	message.append (key)
	message.append (form[key].value)
#
if "host" in form:
	message.append ("host = " + form["host"].value)
#
if "date" in form:
	message.append ("date = " + form["date"].value)
#
if "key" in form:
	message.append ("key = " + form["key"].value)
#
#
file_in = "/var/tmp/count_up/count.txt"
#
str_in = file_to_str_proc (file_in)

nn = int (str_in)
nn += 1
str_out = "%d\n" % nn
file_write_proc (file_in,str_out)

message.append ("%d" % nn)
message.append ('*** count_up.py *** end ***')
#
result = {}
result['message'] = message
result['str_in'] = str_in
result['str_out'] = str_out
#
str_aa = json.dumps (result)
print ("Content-type: text/json; charset=UTF-8\n\n")
print (str_aa)
#
# ---------------------------------------------------------------------

File: examples/api/count_up/count_up.js
// -----------------------------------------------------------------------
//
//	count_up.js
//					Jun/01/2015
//
// -----------------------------------------------------------------------
jQuery (function ()
{
	jQuery ("#outarea_aa").text ("*** count_up.js *** start ***");

	var url_api = "./count_up.py";

	jQuery("#outarea_ee").text ("url_api = " + url_api);

	show_received_data_proc (url_api);

	jQuery ("#outarea_hh").text ("*** count_up.js *** end ***");
});

// -----------------------------------------------------------------------
function show_received_data_proc (url_api)
{
var str_tmp = "*** show_received_data_proc *** bbb ***<br />";
str_tmp += url_api + "<br />";
jQuery("#outarea_bb").html (str_tmp);

	var args = {host : "host_name",
			date: "20150601-155055",
			key: "B0157"};

	jQuery.post (url_api,args,function (data_received)
		{
		var out_str = data_received.str_in;

		jQuery("#result_aa").html (out_str);

		var str_message = "";

		for (var it in data_received.message)
			{
			str_message += data_received.message[it] + "<br />";
			}

		jQuery("#message").html (str_message);
		});
}

// -----------------------------------------------------------------------
